### ISSUE TITLE
Checkpoints + usage of different base BERT model

### ISSUE THREAD

**Issue by zyngielg** [stanford-futuredata/ColBERT#45]
Hello, I would love to use ColBERT in my project as a retriever component in a retriever-reader architecture, hence I would prefer to use the `huggingface` API to create the ColBERT model in the code and load the checkpoints :)  I have three small questions:

1. About the checkpoints of the pretrained on MsMARCO ColBERT. Is it possible to download the pretrained model's weights so I could use them in my implementation? Or the only solution is to use the training script and slightly modify it so I can save the weights for later use.
2. The used BERT model was the `bert-base-uncased`. My project's domain is the medical area, hence I was wondering whether using the training script with a small change - replacing the `bert-base-uncased` with the `biobert` would result in a properly pretrained model. 
3. Is there any plan on creating the implementation of the the ColBERT's workflow available via the `huggingface` library?

Thank you very much for your work 💪🏻

**Comment by littlewine** [comment]
1. You can find a similar version of the model trained by other people [here](https://huggingface.co/vespa-engine/colbert-medium). tbh I am not sure how you can this huggingface pretrained model in a downstream task (ie. index marco or doing retrieval) - in other words whether this is compatible with the existing code repo. It is also a question for me whether we can directly use huggingface models instead of the expected colbert checkpoints that this repo expects during inference (either for retrieval or indexing) @okhat could you help us on that? :) 
2. In this case, you need to train from scratch and make the necessary changes in the code. Briefly, you'd need to go through all of [those](https://github.com/stanford-futuredata/ColBERT/search?q=from_pretrained) code matches and change the input of `from_pretrained` to the model string you want. Surely in the `training.py` script but also possibly in evaluation

**Comment by zyngielg** [comment]
@littlewine thanks for the reply

@1 I found this model, however it has slightly different parameters (linear layer output size and from what I remember the lenght of the embeddings being 512 instead of 768). Still, for now I indeed plan to use that one :) And about the usage in the downstream task - I found it slightly difficult to follow the `retrieve` script's logic and I guess the smartest choice would be to follow the description of the workflow in the paper.

**Comment by okhat** [comment]
Hey folks! Happy to send you a checkpoint if you send me an email. A public/official release will take some time.

There are public checkpoints by other people that are compatible with (and produced from) this code, including that by PyTerrier-ColBERT: https://github.com/terrierteam/pyterrier_colbert --- see the Demo section.

**Comment by jobergum** [comment]
We over at vespa.ai have released a ColBERT weights which is based on a 6 layer MiniLM model with just 22.7M parameters which uses 32 dimensions instead of 128.  So for question 2 you can replace bert-base-uncased with a different model which also uses BERT tokenization. 

You can use https://huggingface.co/vespa-engine/col-minilm and then use the save_weights on it to get a .dnn checkpoint which can be used with more routines (but with similarity=cosine and dim=32). 
(https://github.com/stanford-futuredata/ColBERT/blob/abb5b684e4cda297f3ff58b52e70d5b90270e900/colbert/utils/utils.py#L37)

**Comment by zyngielg** [comment]
Thanks all for your activity in the discussion :)

**Comment by Dopaminezsy** [comment]
Hi @zyngielg,
Now do you have any idea about how to use huggingface API to create the ColBERT model in the code and load the checkpoints?
